File: day2/ques5.cpp
#include <iostream>
using namespace std;

int main() {
    int num, sum = 0;

    cout << "Enter a number: ";
    cin >> num;

    int temp = num;  // Save original number

    while (temp != 0) {
        int lastDigit = temp % 10;  // Extract last digit
        sum += lastDigit;           // Add to sum
        temp = temp / 10;           // Remove last digit
    }

    cout << "Sum of digits of " << num << " = " << sum << endl;

    return 0;
}

File: day2 python/ques5.py
# Take input from the user
num = int(input("Enter a number: "))

# Initialize sum variable
digit_sum = 0

# Store original number for absolute processing
n = abs(num)

# Loop to extract and sum digits
while n > 0:
    digit_sum += n % 10  # Extract the last digit
    n //= 10             # Remove the last digit

print(f"The sum of digits of {num} is: {digit_sum}")
